python/examples/ingestion_with_python_config/telemetry_config.py

```diff
@@ -71,131 +66,166 @@ def nostromos_lv_426() -> TelemetryConfig:
+                "$threshold": 470,
+            },
+            action=RuleActionCreateDataReviewAnnotation(
+                # User in your organization to notify
+                # assignee="ellen.ripley@weylandcorp.com",
+                tags=["nostromo"],
             ),
-            RuleConfig(
-                name="failure",
-                description="Checks for failures reported by logs",
-                expression=named_expressions["log_substring_contains"],
-                channel_references=[
-                    {
-                        "channel_reference": "$1",
-                        "channel_config": log_channel,
-                    },
-                ],
-                sub_expressions={
-                    "$sub_string": "failure",
+        ),
+        RuleConfig(
+            name="failure",
+            description="Checks for failures reported by logs",
+            expression=named_expressions["log_substring_contains"],
+            rule_client_key="failure-rule",
+            asset_names=["NostromoLV426"],
+            channel_references=[
+                {
+                    "channel_reference": "$1",
+                    "channel_config": log_channel,
                 },
-                action=RuleActionCreateDataReviewAnnotation(
-                    # User in your organization to notify
-                    # assignee="ellen.ripley@weylandcorp.com",
-                    tags=["nostromo", "failure"],
-                ),
+            ],
+            sub_expressions={
+                "$sub_string": "failure",
+            },
+            action=RuleActionCreateDataReviewAnnotation(
+                # User in your organization to notify
+                # assignee="ellen.ripley@weylandcorp.com",
+                tags=["nostromo", "failure"],
             ),
-            RuleConfig(
-                name="overvoltage",
-                namespace="voltage",
-                namespace_rules=rule_namespaces,
-                channel_references=[
-                    # INFO: Can use either "channel_identifier" or "channel_config"
-                    {
-                        "channel_reference": "$1",
-                        "channel_identifier": vehicle_state_channel.fqn(),
-                    },
-                    {
-                        "channel_reference": "$2",
-                        "channel_config": voltage_channel,
-                    },
-                ],
+        ),
+        RuleConfig(
+            name="overvoltage",
+            description="Checks for overvoltage while accelerating",
+            expression='$1 == "Accelerating" && $2 > 80',
+            rule_client_key="overvoltage-rule",
+            asset_names=["NostromoLV426"],
+            channel_references=[
+                # INFO: Can use either "channel_identifier" or "channel_config"
+                {
+                    "channel_reference": "$1",
+                    "channel_identifier": vehicle_state_channel.fqn(),
```

Comment:
I'm thinking about how to distill the example to be something much more simple. When I originally wrote the example I tried to show a little bit of everything but that's probably not necessary and may lock us into feature sets that we eventually want to move away from. `components` for example is something we're planning to deprecate.

For this example in particular, rather than `channel_identifier` or `channel_config` what if we do a union type using `channel` as the key which takes in either a string or config?

Also we should probably create a method called `identifier` on `ChannelConfig` and make it an alias for `fqn`.

Reply:
Let me know if what I've pushed aligns with what you were thinking here or if you have an even simpler vision in mind!